contracts/clients/src/Aggregator.ts

```diff
@@ -38,12 +39,18 @@ export type BundleReceiptError = {
   submitError: string | undefined;
 };
 
-export type BundleReceipt = {
-  transactionIndex: number;
-  transactionHash: string;
+/**
+ * The BLS Wallet specific values in a {@link BundleReceipt}.
+ */
+export type BlsWallet = {
   bundleHash: string;
-  blockHash: string;
-  blockNumber: number;
+};
+
+/**
+ * The bundle receipt returned from a BLS Wallet Aggregator instance. It is a combination of an ethers {@link ContractReceipt} and a {@link BlsWallet} type.
+ */
+export type BundleReceipt = ContractReceipt & {
```

Comment:
Been thinking about the validation in this file. I tried adding a property to the return type for `receiptFromBundle` - `invalidProperty: "invalid property”,`. When we’re using our own aggregator, we have confidence the bundleReceipt will be correct, and as mentioned in [this]() comment, our integration tests will pick up certain changes to the returned bundleReceipt. But what if a third party developer is using the client module/provider and decides to use their own aggregator. In this case, invalid responses could reach client module code. 

My question is, should we validate incoming responses so that we don’t allow invalid data into the client module?

Reply:
I think that is probably a good future issue.  Just my opinion but I think validating the contract receipt right now might be a little scope creep.  This is probably something we can handle if people start using other aggregators that don't follow our pattern here.